did_parser/src/lib.rs

```diff
@@ -0,0 +1,16 @@
+extern crate once_cell;
+extern crate regex;
+
+mod error;
+mod parsed_did;
+mod parsed_did_url;
+mod utils;
+
+use std::ops::Range;
+
+type DIDRange = Range<usize>;
```

Comment:
You mean if you decided to use something else instead of a `Range<usize>`? I don't really think it would be a problem. Yeah, it's easier to change it in one place alone, but on the other hand the compiler will just complain if you happen to forget a spot. And it will be easier to see what the actual type is.

But I think this is more of a personal preference than anything else, so feel free to disregard it if it's more convenient for you like that. I generally just alias if the type def is very long 😅

Reply:
Hm. I really do think type alias makes sense in this case. It is a matter of less than second to check the type in any modern IDE.